language-support/scala/codegen/src/main/scala/com/digitalasset/lf/codegen/CodeGen.scala

```diff
@@ -137,49 +124,34 @@ object CodeGen {
   private def combineInterfaces(dar: Dar[Interface]): EnvironmentInterface =
     EnvironmentInterface.fromReaderInterfaces(dar)
 
-  private def combineEnvInterfaces(as: NonEmptyList[EnvironmentInterface]): EnvironmentInterface =
-    as.suml1
+  private def templateCount(interface: EnvironmentInterface): Int =
+    interface.typeDecls.count {
+      case (_, InterfaceType.Template(_, _)) => true
+      case _ => false
+    }
```

Comment:
Moved here from `LFUtil` because it was used only here and didn't depend on any field of `LFUtil`.